hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/MapIndexScanPMigrationStressTest.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.map;
+
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.IndexType;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.jet.sql.SqlTestSupport;
+import com.hazelcast.map.IMap;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.jet.sql.SqlTestSupport.assertRowsOrdered;
+
+public class MapIndexScanPMigrationStressTest extends SimpleTestInClusterSupport {
+    static final int ITEM_COUNT = 750_000;
+    static final int ITEM_TO_LOOKUP = 250_000;
+    static final int COUNT_DIVIDER = 3;
+    static final int MEMBERS_COUNT = 3;
+    static final String MAP_NAME = "map";
+
+    private IMap<Integer, Integer> map;
+
+    @BeforeClass
+    public static void setUpClass() {
+        initializeExceptLast(MEMBERS_COUNT, smallInstanceConfig());
+    }
+
+    @Before
+    public void before() {
+        map = instance().getMap(MAP_NAME);
+    }
+
+    @Test
+    @Ignore // TODO: [sasha] un-ignore after IMDG engine removal
+    public void stressTestPointLookup() {
+        List<SqlTestSupport.Row> expected = new ArrayList<>();
+        for (int i = 0; i <= ITEM_COUNT; i++) {
+            map.put(i, i);
+        }
+        expected.add(new SqlTestSupport.Row(ITEM_TO_LOOKUP, ITEM_TO_LOOKUP));
+
+        IndexConfig indexConfig = new IndexConfig(IndexType.HASH, "this").setName(randomName());
+        map.addIndex(indexConfig);
+
+        MutatorThread mutator = new MutatorThread(instances(), instances().length - 1, 500L);
+        mutator.start();
+
+        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " WHERE this=" + ITEM_TO_LOOKUP, expected);
```

Comment:
With single query, we are rather stress testing cluster membership changes, not the correctness of index based queries.

Reply:
Did I understand correctly : should I multiply count of queries?

Reply:
I think that would increase chances of detecting invalid behaviour.